boto3/session.py

```diff
@@ -306,11 +308,40 @@ def resource(self, service_name, region_name=None, api_version=None,
 
         :return: Subclass of :py:class:`~boto3.resources.base.ServiceResource`
         """
+        try:
+            resource_model = self._loader.load_service_model(
+                service_name, 'resources-1', api_version)
+        except UnknownServiceError as e:
```

Comment:
Do we need to subclass these new exceptions from the exceptions that you now catch for backwards compatibility reasons?